doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,470 @@
+
+.. rubric:: Returns
+
+The digest of the given string as a binary string
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* digest_type - the supported values are the following (based on the OpenSSL version):
+
+  .. list-table::
+      :header-rows: 1
+
+      * - Value Name for OpenSSL 1.0.2
+        - Value Name for OpenSSL 1.1.x addition
+      * - 'MD5'
+        - 'BLAKE2B512'
+      * - 'SHA1'
+        - 'BLAKE2S256'
+      * - 'SHA224'
+        - 'RIPEMD'
+      * - 'SHA256'
+        - 'RMD160'
+      * - 'SHA384'
+        - 'SHAKE128'
+      * - 'SHA512'
+        - 'SHAKE256'
+      * - 'MD4'
+        - 'SM3'
+      * - 'RIPEMD160'
+        - 'WHIRLPOOL'
+
+* str - String used to generate the digest string.
+
+.. _encryption_variables:
+
+.. rubric:: Encryption threshold variables
+
+The maximum key length limits are defined by OpenSSL. Server administrators can limit the maximum key length using the encryption threshold variables. 
+
+The variables are automatically registered when `component_encryption_udf` is installed.
+
+
+.. list-table::
+   :header-rows: 1
+
+   * - Variable Name
+   * - :ref:`dh_threshold`
+
+.. _dh_threshold:
+
+.. rubric:: ``encryption_udf.dh_bits_threshold``
+
+The variable sets the maximum limit for the :ref:`create_dh_parameters <create_dh_param>` user-defined function and takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 10000
+
+The range for this variable is from 1024 to 10,000. The default value is 10,000.
+
+.. _dsa_threshold:
+
+.. rubric:: `encryption_udf.dsa_bits_threshold`
+
+The variable sets the threshold limits for :ref:`create_asymmetric_priv_key <create_priv_key>` user-defined function when the function is invoked with the `DSA` parameter and takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 9984
+
+
+The range for this variable is from 1,024 to 9,984. The default value is 9,984.
+
+.. _rsa_threshold:
+
+.. rubric:: `encryption_udf.rsa_bits_threshold`
+
+The variable sets the threshold limits for the :ref:`create_asymmetric_priv_key <create_priv_key>` user-defined function when the function is invoked with the `RSA` parameter and takes precedence over the OpenSSL maximum length value.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 16384
+
+The range for this variable is from 1,024 to 16,384.
```

Comment:
Just for consistency reasons with previous variables.
```suggestion
The range for this variable is from 1,024 to 16,384. The default value is 16,384.
```

Reply:
Done.